fingerprint_verification.py

```diff
@@ -4,41 +4,117 @@
 # Any extensions done:
 ######################################################################
 
+# Mileston #1
 def read_print(filename):
-    """Reads the given print and returns a dictionary with all info."""
-    # Your turn!
-
-
-
-
-
-
-# You can add your various check functions below
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-if __name__ == '__main__':
-    # Just to get you started
-    data = read_print("./prints/User1_Original.txt")
-    print(data)
-
-    # Add anything else you need for testing down here as you go!
+    with open(filename, 'r') as file:
+        lines = file.readlines()
+
+    # Extract the name, width, and height
+    name = lines[0].strip()
+    width = int(lines[1].strip())
+    height = int(lines[2].strip())
+
+    # Extract the fingerprint data
+    fingerprint = [line[:-1] for line in lines[3:]]
+
+    # Verify the dimensions of the fingerprint
+    # for line in fingerprint:
+    #     if len(line) != width:
+    #         raise ValueError("Fingerprint dimensions do not match the specified width")
+
+    # Create the data dictionary
+    data = {
+        'name': name,
+        'width': width,
+        'height': height,
+        'fingerprint': fingerprint
+    }
+
+    # print(data)
+        # Confirmed it is printing the contents in dictionary format
+    return data
+
+# Milestone #2
+def simple_check(fingerprint1, fingerprint2):
+    return fingerprint1['fingerprint'] == fingerprint2['fingerprint']
+
+# Milestone #3
+def variant_check(fingerprint1, fingerprint2, threshold=95.0):
+    if fingerprint1['width'] != fingerprint2['width'] or fingerprint1['height'] != fingerprint2['height']:
+        return False
+
+    match_count = 0
+    total_pixels = 0
+
+    for i in range(7, fingerprint1['height']):
+        for j in range(fingerprint1['width']):
+            if j < len(fingerprint1['fingerprint'][i]) and j < len(fingerprint2['fingerprint'][i]):
```

Comment:
Were you getting index out of bounds here? Such that your fingerprint structure did not agree with the posted height and width? That shouldn't have happened.